### PR TITLE
Upgrade prometheus-client-mmap

### PR DESCRIPTION
**Pull request by errm**

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -5,12 +5,16 @@ require "rubocop/rake_task"
 RSpec::Core::RakeTask.new(:spec)
 RuboCop::RakeTask.new
 
-task default: %i(spec rubocop)
+task default: %i(spec rubocop integration)
```
Comment (davidstosik) [inline Rakefile:8]: Out of curiosity, why does this need a new type of test, that does not fit in `spec`?

**Review by davidstosik — COMMENTED** [review]
@errm Thanks for working on this!

As far as my limited knowledge around this allows me to judge, this looks good. 👍🏻

File: Rakefile (hunk 1 above)
Comment (errm) [reply Rakefile:8]: So in the spec helper I stub/monkey patched a lot of the lower level parts of prometheus-client-mmap, specifically writing stuff to the file system and added a way to reset everything between each test.

I wanted to add an integration test where there is no patching, and we test that we can record metrics in on process (i.e. a rails application, simulated here by this test script) and export them using the sidecar process.

I suspect if I was a little less rusty about how rspec works I could have got it to do all this, but I didn't have a free day to work out what to do, so just writing a script that starts a process running the exporter and exercises a few of the examples from the README was the best I could come up with in the time I had free 🤷

**Comment by robertomiranda** [comment]
@errm any plans of getting this change merged?

**Pull request merged**